crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,332 @@
+    pub fn is_finite(self) -> bool {
+        self.sin.is_finite() && self.cos.is_finite()
+    }
+
+    /// Returns `true` if the rotation is NaN.
+    #[inline]
+    pub fn is_nan(self) -> bool {
+        self.sin.is_nan() && self.cos.is_nan()
+    }
+
+    /// Returns `true` if the rotation is near [`Rotation2d::IDENTITY`].
+    #[inline]
+    pub fn is_near_identity(self) -> bool {
+        // TODO: We might be able to use the rotation sine and cosine
+        //       directly instead of first converting to radians.
+        // Same as `Quat::is_near_identity`
+        let threshold_angle = 0.002_847_144_6;
+        self.as_radians().abs() < threshold_angle
+    }
+
+    /// Returns the angle in radians needed to make `self` and `other` coincide.
+    #[inline]
+    pub fn angle_between(self, other: Self) -> f32 {
+        (other * self.inverse()).as_radians()
+    }
+
+    /// Returns the inverse of the rotation. This is also the conjugate
+    /// of the unit complex number representing the rotation.
+    #[inline]
+    #[must_use]
+    #[doc(alias = "conjugate")]
+    pub fn inverse(self) -> Self {
+        Self {
+            cos: self.cos,
+            sin: -self.sin,
+        }
+    }
+
+    /// Performs a linear interpolation between `self` and `rhs` based on
+    /// the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `rhs`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::IDENTITY;
+    /// let rot2 = Rotation2d::from_radians(std::f32::consts::FRAC_PI_2);
+    ///
+    /// let result = rot1.lerp(rot2, 0.5);
+    ///
+    /// assert_eq!(result.as_radians(), std::f32::consts::FRAC_PI_4);
+    /// ```
+    #[inline]
+    pub fn lerp(self, end: Self, s: f32) -> Self {
+        Self::from_sin_cos(self.sin.lerp(end.sin, s), self.cos.lerp(end.cos, s))
+    }
+
+    /// Performs a spherical linear interpolation between `self` and `end`
+    /// based on the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `end`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Rotation2d;
+    /// #
+    /// let rot1 = Rotation2d::from_radians(std::f32::consts::FRAC_PI_4);
+    /// let rot2 = Rotation2d::from_radians(-std::f32::consts::PI);
+    ///
+    /// let result = rot1.slerp(rot2, 1.0 / 3.0);
+    ///
+    /// assert_eq!(result.as_radians(), std::f32::consts::FRAC_PI_2);
+    /// ```
+    #[inline]
+    pub fn slerp(self, end: Self, s: f32) -> Self {
+        let delta = end * self.inverse();
+        self * Self::from_radians(delta.as_radians() * s)
+    }
+}
+
+impl From<f32> for Rotation2d {
+    /// Creates a [`Rotation2d`] from a counterclockwise angle in radians.
+    fn from(rotation: f32) -> Self {
+        Self::from_radians(rotation)
+    }
+}
```

Comment:
I'm not very fan of this: `f32` doesn't necessarily represent an angle in radians.
I would prefer the user to explicitly use `from_radians` or `from_degrees` instead.

Reply:
I think this is mostly for backwards compatibility, to allow a you to still just pass on radians everywhere you could before. IMO that's a used feature.

Reply:
A while ago I opened some discussion on adding an `Angle` type or `Radians` and `Degrees` types: #11132. The overall consensus seemed to be that Bevy should use radians for rotations everywhere, and you should still be able to pass implicit floats too.

Not having this impl would also make this a breaking change, and a lot of APIs would be a bit longer.

Reply:
I agree this is a desirable trade-off. Radians are very common in all sorts of angle math and demanding an explicit conversion everywhere would be too verbose.

I would however consider explaining radians in beginner-oriented documentation/examples, because those are the most likely to expect degree-based APIs.